### PR TITLE
feat(core): add fallback values per flag

### PR DESCRIPTION
**Pull request by cstrnt**

### CODE REVIEW

File: packages/core/src/index.ts [hunk 1]
```diff
@@ -252,10 +255,19 @@ export class Abby<
...
-    if (!storedValue && defaultValue != null) {
-      return defaultValue;
+    const fallbackValue = this._cfg.settings?.flags?.fallbackValues?.[key];
+
+    // if we dont have a stored value
+    // we first check if we have a specified fallback value for this flag
+    // otherwise we check if we have a specified default value for this flag type
+    // otherwise we return undefined
+    if (!storedValue) {
+      if (fallbackValue != null) {
```
Comment (slaesh) [inline packages/core/src/index.ts:265]: still not sure why these values arent allowed? we could easily use `key in obj`?

**Review by slaesh — APPROVED** [review]
it would be cool to have only "one object" which describes my flags.. all in one, name, value and type.
and this would be the "now known as fallback-values" thingy.

there is a value which the user has chosen, we could infer or compile the type and there would be no need for the other two objects (type and default)..

but its better than before :+1: 

since default and fallback sounds similar, guess we need some docs stating out what it what

**Comment by cstrnt** [comment]
> it would be cool to have only "one object" which describes my flags.. all in one, name, value and type. and this would be the "now known as fallback-values" thingy.
> 
> there is a value which the user has chosen, we could infer or compile the type and there would be no need for the other two objects (type and default)..
> 
> but its better than before 👍
> 
> since default and fallback sounds similar, guess we need some docs stating out what it what

Something like ?


```ts
{
  // other config
  flags: {
    myFlag: {
       type: "Boolean",
       fallbackValue: false
    }
  }
}
```

**Comment by slaesh** [comment]
why not just 
```ts
{
  flags: {
    myFlag: false
    serverTimeout: 3000
  }
}
```

**Comment by cstrnt** [comment]
> why not just
> 
> ```ts
> {
>   flags: {
>     myFlag: false
>     serverTimeout: 3000
>   }
> }
> ```
> 
> but definitely better than having two separate ones

I honestly feel like that this is not really clear that it's the fallback / default value. For me it reads like this is the value to use. Might be mistaken though!

**Comment by slaesh** [comment]
in my mind IT IS the value to use.. until the backend (maybe) sends another value though :)
thats the beauty of reactive apps ;) but its already approved anyways

**Pull request merged**